src/hacbs/utils/commits-utils.ts

```diff
@@ -1,59 +1,75 @@
 import { PipelineRunLabel } from '../consts/pipelinerun';
 import { PipelineRunKind, Commit } from '../types';
 
+export const createCommitObjectFromPLR = (plr: PipelineRunKind): Commit => {
+  if (!plr || !plr?.metadata.labels?.[PipelineRunLabel.COMMIT_LABEL]) {
+    return null;
+  }
+  const commitSHA = plr.metadata.labels?.[PipelineRunLabel.COMMIT_LABEL];
+  const commitBranch = plr.metadata.annotations?.[PipelineRunLabel.COMMIT_BRANCH_ANNOTATION] ?? '';
+  const commitUser = plr.metadata.labels?.[PipelineRunLabel.COMMIT_USER_LABEL];
+  const shaURL = plr.metadata.annotations?.[PipelineRunLabel.COMMIT_URL_ANNOTATION];
+  const shaTitle = plr.metadata.annotations?.[PipelineRunLabel.COMMIT_SHA_TITLE_ANNOTATION];
+  const creationTime = plr.metadata.creationTimestamp;
+  const application = plr.metadata.labels[PipelineRunLabel.APPLICATION];
+  const component = plr.metadata.labels[PipelineRunLabel.COMMIT_COMPONENT_LABEL] ?? '';
+  const repoURL = plr.metadata.labels[PipelineRunLabel.COMMIT_REPO_URL_LABEL];
+  const repoOrg = plr.metadata.labels[PipelineRunLabel.COMMIT_REPO_ORG_LABEL];
+  const gitProvider = plr.metadata.labels[PipelineRunLabel.COMMIT_PROVIDER_LABEL];
+
+  return {
+    metadata: {
+      uid: commitSHA,
+      name: commitSHA,
+    },
+    components: [component],
+    user: commitUser,
+    sha: commitSHA,
+    displayName: commitSHA.slice(0, 7),
+    shaURL,
+    repoURL,
+    repoOrg,
+    gitProvider,
+    branch: commitBranch,
+    creationTime,
+    pipelineRuns: [plr],
+    application,
+    shaTitle,
+  };
+};
+
+const updateCommitObject = (plr: PipelineRunKind, commit: Commit): Commit => {
+  const newCommit = commit;
+  const creationTime = plr.metadata.creationTimestamp;
+  const application = plr.metadata.labels[PipelineRunLabel.APPLICATION];
+  const component = plr.metadata.labels[PipelineRunLabel.COMPONENT] ?? '';
+
+  if (newCommit.creationTime < creationTime) {
+    newCommit.creationTime = creationTime;
+  }
+  if (!newCommit.application) {
+    newCommit.application = application;
+  }
```

Comment:
I don't think this should ever happen as we're always working within the context of a single application.

Reply:
removed